File: Python/268.py
# Initial thoughts
    # Can probably use XOR b/c if you flip the bits once for each val in nums
        # and once for each val in range(n+1), the missing number will be the remainder
    # Need to start at 0
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        val = 0
        for i in range(len(nums)+1): val ^= i
        for i in nums: val ^= i
        return val

File: C++/268.cpp
/*
- Use the fact that xXORx = 0
- In particular, XOR 0 by each number in the array and then each number 1-N
*/
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int result = 0;
        for(int i = 0; i < nums.size(); i++){
            result ^= nums[i];
        }
        for(int i = 0; i < nums.size() + 1; i++){
            result ^= i;
        }
        return result;
    }
};